frame/vamm/src/lib.rs

```diff
@@ -581,11 +581,13 @@ pub mod pallet {
 				.ok_or(ArithmeticError::Overflow)?;
 
 			let new_input_amount = match direction {
-				Direction::Add =>
-					input_asset_amount.checked_add(swap_amount).ok_or(ArithmeticError::Overflow)?,
+				Direction::Add => {
+					input_asset_amount.checked_add(swap_amount).ok_or(ArithmeticError::Underflow)?
+				},
```

Comment:
Are you adding positive numbers? In that case, it should be `Overflow` in the error case